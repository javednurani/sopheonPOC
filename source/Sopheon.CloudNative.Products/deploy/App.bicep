param sites_StratusProductManagement_Dev_name string = 'StratusProductManagement-Dev'
param serverfarms_StratusProductManagement_AppPlan_Dev_externalid string = '/subscriptions/1c4bef1d-8a40-4a6d-96d6-764bb466ac46/resourceGroups/Accolade-Next-Gen/providers/Microsoft.Web/serverfarms/StratusProductManagement-AppPlan-Dev'

resource sites_StratusProductManagement_Dev_name_resource 'Microsoft.Web/sites@2021-02-01' = {
  name: sites_StratusProductManagement_Dev_name
  location: 'West US'
  tags: {
    'Environment Type': 'Research'
    Owner: 'CloudTeam-1'
    'Review Date': '06-30-21'
  }
  kind: 'app'
  properties: {
    enabled: true
    hostNameSslStates: [
      {
        name: 'stratusproductmanagement-dev.azurewebsites.net'
        sslState: 'Disabled'
        hostType: 'Standard'
      }
      {
        name: 'stratusproductmanagement-dev.scm.azurewebsites.net'
        sslState: 'Disabled'
        hostType: 'Repository'
      }
    ]
    serverFarmId: serverfarms_StratusProductManagement_AppPlan_Dev_externalid
    reserved: false
    isXenon: false
    hyperV: false
    siteConfig: {
      numberOfWorkers: 1
      acrUseManagedIdentityCreds: false
      alwaysOn: true
      http20Enabled: false
      functionAppScaleLimit: 0
      minimumElasticInstanceCount: 1
      appSettings: {
        
      }
    }
    scmSiteAlsoStopped: false
    clientAffinityEnabled: true
    clientCertEnabled: false
    clientCertMode: 'Required'
    hostNamesDisabled: false
    customDomainVerificationId: 'F50C4A4633DCAF18B0C09B554884823C47C074E58674E017191D38260A145C6A'
    containerSize: 0
    dailyMemoryTimeQuota: 0
    httpsOnly: false
    redundancyMode: 'None'
    storageAccountRequired: false
    keyVaultReferenceIdentity: 'SystemAssigned'
  }
}

resource sites_StratusProductManagement_Dev_name_ftp 'Microsoft.Web/sites/basicPublishingCredentialsPolicies@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'ftp'
  properties: {
    allow: true
  }
}

resource sites_StratusProductManagement_Dev_name_scm 'Microsoft.Web/sites/basicPublishingCredentialsPolicies@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'scm'
  properties: {
    allow: true
  }
}

resource sites_StratusProductManagement_Dev_name_web 'Microsoft.Web/sites/config@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'web'
  properties: {
    numberOfWorkers: 1
    defaultDocuments: [
      'Default.htm'
      'Default.html'
      'Default.asp'
      'index.htm'
      'index.html'
      'iisstart.htm'
      'default.aspx'
      'index.php'
      'hostingstart.html'
    ]
    netFrameworkVersion: 'v5.0'
    requestTracingEnabled: true
    requestTracingExpirationTime: '12/31/9999 11:59:00 PM'
    remoteDebuggingEnabled: false
    remoteDebuggingVersion: 'VS2019'
    httpLoggingEnabled: true
    acrUseManagedIdentityCreds: false
    logsDirectorySizeLimit: 35
    detailedErrorLoggingEnabled: true
    publishingUsername: '$StratusProductManagement-Dev'
    scmType: 'None'
    use32BitWorkerProcess: true
    webSocketsEnabled: false
    alwaysOn: true
    managedPipelineMode: 'Integrated'
    virtualApplications: [
      {
        virtualPath: '/'
        physicalPath: 'site\\wwwroot'
        preloadEnabled: true
      }
    ]
    loadBalancing: 'LeastRequests'
    experiments: {
      rampUpRules: []
    }
    autoHealEnabled: false
    vnetRouteAllEnabled: false
    vnetPrivatePortsCount: 0
    localMySqlEnabled: false
    ipSecurityRestrictions: [
      {
        ipAddress: 'Any'
        action: 'Allow'
        priority: 1
        name: 'Allow all'
        description: 'Allow all access'
      }
    ]
    scmIpSecurityRestrictions: [
      {
        ipAddress: 'Any'
        action: 'Allow'
        priority: 1
        name: 'Allow all'
        description: 'Allow all access'
      }
    ]
    scmIpSecurityRestrictionsUseMain: false
    http20Enabled: false
    minTlsVersion: '1.2'
    scmMinTlsVersion: '1.0'
    ftpsState: 'AllAllowed'
    preWarmedInstanceCount: 0
    functionAppScaleLimit: 0
    healthCheckPath: '/swagger/ui'
    functionsRuntimeScaleMonitoringEnabled: false
    minimumElasticInstanceCount: 1
    azureStorageAccounts: {}
  }
}

resource sites_StratusProductManagement_Dev_name_sites_StratusProductManagement_Dev_name_azurewebsites_net 'Microsoft.Web/sites/hostNameBindings@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: '${sites_StratusProductManagement_Dev_name}.azurewebsites.net'
  properties: {
    siteName: 'StratusProductManagement-Dev' //TODO: Configure for TokenReplace
    hostNameType: 'Verified'
  }
}

resource sites_StratusProductManagement_Dev_name_Microsoft_AspNetCore_AzureAppServices_SiteExtension 'Microsoft.Web/sites/siteextensions@2021-02-01' = {
  parent: sites_StratusProductManagement_Dev_name_resource
  name: 'Microsoft.AspNetCore.AzureAppServices.SiteExtension'
}
