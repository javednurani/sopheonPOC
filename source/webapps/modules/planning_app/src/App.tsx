import React from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { AppProps } from './TempSopheonShellAPI';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const App: React.FunctionComponent<Props> = () => <div></div>;

export default App;
