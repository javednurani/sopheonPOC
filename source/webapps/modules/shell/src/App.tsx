import { initializeIcons, registerIcons, ScrollablePane, ScrollbarVisibility, Stack } from '@fluentui/react';
import { useTheme } from '@fluentui/react-theme-provider';
import React, { CSSProperties, FunctionComponent } from 'react';
import { useIntl } from 'react-intl';
import { BrowserRouter, Route, Switch } from 'react-router-dom';

import IdleMonitor from './authentication/IdleMonitor';
import Login from './authentication/Login';
import Signup from './authentication/Signup';
import { DynamicModule } from './DynamicModule';
import Footer from './footer/Footer';
import Header from './header/Header';
import { ReactComponent as LLogo } from './images/Lucy24_logo.svg';
import { appModules } from './settings/appModuleSettings';
import { shellApi } from './ShellApi';
import { ChangeThemeAction } from './themes/themeReducer/themeReducer';

export interface AppProps {
  changeTheme: (useDarkTheme: boolean) => ChangeThemeAction;
}

const App: FunctionComponent<AppProps> = ({ changeTheme }: AppProps) => {
  const { formatMessage } = useIntl();
  const loadingMessage: string = formatMessage({ id: 'fallback.loading' });
  useTheme();

  initializeIcons();

  const lucyIconStyle: CSSProperties = {
    width: '20px',
    height: '20px',
    overflow: 'visible',
  };

  registerIcons({
    icons: {
      Lucy: <LLogo style={lucyIconStyle} />,
    },
  });

  const appStyle: CSSProperties = {
    textAlign: 'center',
    height: '100vh',
  };

  const pageContainerStyle: CSSProperties = {
    position: 'relative',
    height: '100%',
  };

  // TODO: check for user product existance CLOUD-2148
  const userHasProduct = false;

  return (
    <div className="App" style={appStyle}>
      <BrowserRouter>
        <Switch>
          <Route exact path="/login">
            <Login />
          </Route>
          <Route exact path="/signup">
            <Signup />
          </Route>
          <Route path="/">
            <Stack
              grow
              styles={{
                root: {
                  height: '100%',
                  width: '100%',
                },
              }}
            >
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Header changeTheme={changeTheme} />
                </Stack.Item>
              }
              <Stack.Item shrink>
                <IdleMonitor />
              </Stack.Item>
              <Stack.Item
                verticalFill
                styles={{
                  root: {
                    height: '100%',
                    overflow: 'auto',
                  },
                }}
              >
                <main style={pageContainerStyle} role="main">
                  <ScrollablePane scrollbarVisibility={ScrollbarVisibility.auto}>
                    <Switch>
                      {appModules.map(appModule => (
                        <Route exact key={appModule.scope} path={appModule.routeName}>
                          <DynamicModule module={appModule} loadingMessage={loadingMessage} shellApi={shellApi} />
                        </Route>
                      ))}
                    </Switch>
                  </ScrollablePane>
                </main>
              </Stack.Item>
              {(!location.pathname.includes('product') || userHasProduct) &&
                <Stack.Item>
                  <Footer />
                </Stack.Item>
              }
            </Stack>
          </Route>
        </Switch>
      </BrowserRouter>
    </div>
  );
};

export default App;
