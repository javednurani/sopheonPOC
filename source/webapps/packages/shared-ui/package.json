{
  "name": "@sopheon/shared-ui",
  "version": "1.0.0",
  "description": "A package of UI Components for Sopheon",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "test": ""
  },
  "keywords": [],
  "author": "Sopheon",
  "license": "ISC",
  "dependencies": {
    "@fluentui/react": "^7.155.4",
    "@fluentui/react-theme-provider": "^0.18.5",
    "@microsoft/applicationinsights-react-js": "^3.1.0",
    "@microsoft/applicationinsights-web": "^2.6.1",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "react-intl": "^5.13.4",
    "typescript": "^4.2.2"
  },
  "devDependencies": {
    "@types/node": "^15.0.1",
    "@typescript-eslint/eslint-plugin": "^4.14.0",
    "@typescript-eslint/parser": "^4.14.0",
    "babel-eslint": "^10.1.0",
    "core-js": "^3.9.1",
    "eslint": "^7.14.0",
    "eslint-config-prettier": "6.11.0",
    "eslint-config-sopheon": "^1.0.7",
    "eslint-plugin-html": "^6.1.2",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-prettier": "^3.3.1",
    "eslint-plugin-react": "^7.20.0",
    "eslint-plugin-react-hooks": "^4.0.4",
    "eslint-plugin-simple-import-sort": "^7.0.0",
    "prettier": "^2.0.5",
    "prettier-eslint": "^11.0.0",
    "stylelint": "^13.6.1",
    "stylelint-config-prettier": "^8.0.2",
    "stylelint-config-sass-guidelines": "^7.0.0",
    "stylelint-order": "^4.1.0",
    "stylelint-prettier": "^1.1.2"
  }
}
