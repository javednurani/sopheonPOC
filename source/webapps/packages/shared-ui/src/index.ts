export * from './errorBoundary/ErrorBoundary';
export * from './language/constants';
export * from './language/languages';
export * from './language/messages';
export * from './language/types';
export * from './theme/themes';
