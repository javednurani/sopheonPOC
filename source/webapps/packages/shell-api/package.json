{
  "name": "@sopheon/shell-api",
  "version": "1.0.0",
  "description": "A module federation Shell API package for Sopheon",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "scripts": {
    "build": "tsc",
    "test": "jest --passWithNoTests",
    "test:coverage": "jest --passWithNoTests --coverage",
    "test:watch": "jest --passWithNoTests --watchAll"
  },
  "keywords": [],
  "author": "Sopheon",
  "license": "ISC",
  "dependencies": {
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "react-redux": "^7.2.3",
    "redux": "^4.0.5",
    "redux-devtools-extension": "^2.13.9",
    "redux-saga": "^1.1.3",
    "typescript": "^4.2.2"
  },
  "devDependencies": {
    "@babel/preset-env": "^7.14.0",
    "@babel/preset-react": "^7.13.13",
    "@babel/preset-typescript": "^7.13.0",
    "@types/jest": "^26.0.23",
    "@types/react": "^17.0.3",
    "jest": "^26.6.3"
  }
}
