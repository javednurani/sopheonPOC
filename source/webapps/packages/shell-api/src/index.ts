export * from './store/types';
export * from './store/actions';
export * from './IShellApi';
export * from './IsolatedShellApi';
