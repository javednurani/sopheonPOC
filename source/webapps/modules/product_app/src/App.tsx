import { registerIcons, Stack } from '@fluentui/react';
import { AppProps } from '@sopheon/shell-api';
import React, { useEffect } from 'react';

import { AppDispatchProps, AppStateProps } from './AppContainer';
import { ReactComponent as AeroIndustry } from './images/industryico_Aero.svg';
import { ReactComponent as AgIndustry } from './images/industryico_Ag.svg';
import { ReactComponent as AutoIndustry } from './images/industryico_Auto.svg';
import { ReactComponent as ConstREIndustry } from './images/industryico_ConstRE.svg';
import { ReactComponent as ConsumerIndustry } from './images/industryico_Consumer.svg';
import { ReactComponent as EdIndustry } from './images/industryico_Ed.svg';
import { ReactComponent as EnergyIndustry } from './images/industryico_Energy.svg';
import { ReactComponent as FinIndustry } from './images/industryico_Fin.svg';
import { ReactComponent as GovtIndustry } from './images/industryico_Govt.svg';
import { ReactComponent as HealthIndustry } from './images/industryico_Health.svg';
import { ReactComponent as HospIndustry } from './images/industryico_Hosp.svg';
import { ReactComponent as IndusIndustry } from './images/industryico_Indus.svg';
import { ReactComponent as MediaIndustry } from './images/industryico_Media.svg';
import { ReactComponent as MemberIndustry } from './images/industryico_Member.svg';
import { ReactComponent as ServicesIndustry } from './images/industryico_Services.svg';
import { ReactComponent as TechIndustry } from './images/industryico_Tech.svg';
import { ReactComponent as TeleIndustry } from './images/industryico_Tele.svg';
import { ReactComponent as TransIndustry } from './images/industryico_Trans.svg';
import OnboardingInfo from './onboardingInfo';
import { EnvironmentScopedApiRequestDto } from './types';

export type Props = AppProps<AppStateProps, AppDispatchProps>;

const svgIconStyle: React.CSSProperties = {
  width: '20px',
  height: '20px',
  overflow: 'visible',
};
registerIcons({
  icons: {
    AgIndustryIcon: <AgIndustry style={svgIconStyle} />,
    AeroIndustryIcon: <AeroIndustry style={svgIconStyle} />,
    AutoIndustryIcon: <AutoIndustry style={svgIconStyle} />,
    ConstREIndustryIcon: <ConstREIndustry style={svgIconStyle} />,
    ConsumerIndustryIcon: <ConsumerIndustry style={svgIconStyle} />,
    EduIndustryIcon: <EdIndustry style={svgIconStyle} />,
    EnergyIndustryIcon: <EnergyIndustry style={svgIconStyle} />,
    FinIndustryIcon: <FinIndustry style={svgIconStyle} />,
    GovtIndustryIcon: <GovtIndustry style={svgIconStyle} />,
    HealthIndustryIcon: <HealthIndustry style={svgIconStyle} />,
    HospIndustryIcon: <HospIndustry style={svgIconStyle} />,
    IndusIndustryIcon: <IndusIndustry style={svgIconStyle} />,
    MediaIndustryIcon: <MediaIndustry style={svgIconStyle} />,
    MemberIndustryIcon: <MemberIndustry style={svgIconStyle} />,
    ServicesIndustryIcon: <ServicesIndustry style={svgIconStyle} />,
    TechIndustryIcon: <TechIndustry style={svgIconStyle} />,
    TeleIndustryIcon: <TeleIndustry style={svgIconStyle} />,
    TransIndustryIcon: <TransIndustry style={svgIconStyle} />,
  },
});

const App: React.FunctionComponent<Props> = ({
  currentStep,
  createProduct,
  updateProduct,
  getProducts,
  environmentKey,
  getAccessToken,
  accessToken,
}: Props) => {
  useEffect(() => {
    // getAccessToken triggers Shell action to store access token, freshly acquired from MSAL, in Redux state
    // after getAccessToken is called (here, on ProductApp render), shellApi::accessToken should be up-to-date access token from MSAL.acquireTokenSilent()
    getAccessToken();

    // get any Products for User on initial load of Product App (will determine Onboarding State, ____, etc)
    const requestDto: EnvironmentScopedApiRequestDto = {
      EnvironmentKey: environmentKey,
      AccessToken: accessToken,
    };

    getProducts(requestDto);
  }, []);

  return (
    <div>
      <Stack>
        <OnboardingInfo
          currentStep={currentStep}
          createProduct={createProduct}
          updateProduct={updateProduct}
          environmentKey={environmentKey}
          accessToken={accessToken}
        />
      </Stack>
    </div>
  );
};
export default App;
