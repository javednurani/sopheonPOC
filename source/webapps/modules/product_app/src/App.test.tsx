import { messages } from '@sopheon/shared-ui';
import { FetchStatus } from '@sopheon/shell-api';
import { mount } from 'enzyme';
import { axe, toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { IntlProvider } from 'react-intl';

import App, { Props } from './App';
import OnboardingInfo from './onboarding/onboardingInfo';

expect.extend(toHaveNoViolations);

describe('Testing the App component', () => {
  it('Render test for the App component', () => {
    const appProps: Props = {
      currentStep: 1,
      nextStep: jest.fn(),
      environmentKey: 'asdf',
      getAccessToken: jest.fn(),
      accessToken: '',
      showHeader: jest.fn(),
      hideHeader: jest.fn(),
      products: [],
      getProductsFetchStatus: FetchStatus.DoneSuccess,
      getProducts: jest.fn(),
      createProduct: jest.fn(),
      updateProduct: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App {...appProps} />
      </IntlProvider>
    );

    expect(wrapper.find(OnboardingInfo)).toHaveLength(1);
  });
  it('Accessibility test for the App component', async () => {
    const appProps: Props = {
      currentStep: 1,
      nextStep: jest.fn(),
      environmentKey: 'asdf',
      getAccessToken: jest.fn(),
      accessToken: '',
      showHeader: jest.fn(),
      hideHeader: jest.fn(),
      products: [],
      getProductsFetchStatus: FetchStatus.NotActive,
      getProducts: jest.fn(),
      createProduct: jest.fn(),
      updateProduct: jest.fn(),
    };

    const wrapper = mount(
      <IntlProvider locale="en" messages={messages.en}>
        <App {...appProps} />
      </IntlProvider>
    );

    const results = await axe(wrapper.getDOMNode(), {
      rules: {
        // These axe rules are fixed when app is run in the shell
        region: { enabled: false },
      },
    });
    expect(results).toHaveNoViolations();
  });
});
