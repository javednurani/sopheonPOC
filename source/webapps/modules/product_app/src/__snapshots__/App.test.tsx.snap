// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App component App snapshot render test 1`] = `
<div>
  <label
    className="ms-Label root-76"
  >
    welcome to app
    <div
      className="ms-Stack step1 css-77"
    >
      step1
      Array [
        <button
          aria-label="next"
          className="ms-Button ms-Button--primary root-78"
          data-is-focusable={true}
          onClick={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          type="button"
        >
          <span
            className="ms-Button-flexContainer flexContainer-79"
            data-automationid="splitbuttonprimary"
          >
            <span
              className="ms-Button-textContainer textContainer-80"
            >
              <span
                className="ms-Button-label label-82"
                id="id__6"
                key="id__6"
              >
                next
              </span>
            </span>
          </span>
        </button>,
        "",
      ]
    </div>
  </label>
</div>
`;
